avalon/pipeline.py

```diff
@@ -720,6 +720,29 @@ def get_representation_context(representation):
     return context
 
 
+def get_work_directory():
```

Comment:
Must say I'm not a fan of these getter-style functions..

Reply:
Would be happy to discuss alternatives.

Reply:
To me, the only reason to have a getter is to compute something when returning it. In our case, I think that should rarely if ever be the case, because implicit computations are difficult to debug and easy to misuse. It's too easy to use amongst calls that don't take long to compute, and then suddenly we've got a really slow code base with little idea of where time is being spent. A better alternative is making heavy calls explicit, to make it more difficult to use them, so that they stand out.

In this particular case, my suggestion would be the above; to update the computed version in Session directly.